### ISSUE TITLE
disconnecting the disc during resizing pvc takes about an hour

### ISSUE THREAD

**Issue by elkh510** [Azure/AKS#2260]
**What happened**:
when expanding the size of the pvc, disconnecting the disk from the node takes more than an hour

**What you expected to happen**:
expanding the size of the pvc took less than 5 minutes

**How to reproduce it (as minimally and precisely as possible)**:
```test@(aks-cluster):~$ date
Fri 09 Apr 2021 14:48:42 PM EEST

test@(aks-cluster):~$ kubectl create ns pvc
namespace/pvc created

test@(aks-cluster):~$ kubectl -n pvc apply -f /tmp/pvc_expand.yaml 
persistentvolumeclaim/azure-disk-pvc created

deployment.apps/ubuntu created

test@(aks-cluster):~$ kubectl -n pvc get pvc
NAME             STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS        AGE
azure-disk-pvc   Bound    pvc-0d0dda5b-bd13-470c-bff6-3ec94a64253d   2Gi        RWO            azure-disk          20s
  

test@(aks-cluster):~$ kubectl -n pvc apply -f /tmp/pvc.yaml
test@(aks-cluster):~$ kubectl -n pvc get pvc
NAME             STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS        AGE
azure-disk-pvc   Bound    pvc-0d0dda5b-bd13-470c-bff6-3ec94a64253d   2Gi        RWO            azure-disk          79s


test@(aks-cluster):~$ kubectl -n pvc describe pvc azure-disk-pvc | grep Events -A40
Events:
  Type     Reason                 Age                     From                         Message
  ----     ------                 ----                    ----                         -------
  Warning  VolumeResizeFailed     9m10s (x13 over 9m31s)  volume_expand                error expanding volume "cvp/azure-disk-pvc" of plugin "kubernetes.io/azure-disk": azureDisk - disk resize is only supported on Unattached disk, current disk state: Attached, already attached to  /subscriptions/111111111-1111-111-11111-1111111/resourceGroups/MC_rg-test_test/providers/Microsoft.Compute/virtualMachineScaleSets/aks-default-31009192-vmss/virtualMachines/aks-default-31009192-vmss_0

test@(aks-cluster):~$ kubectl -n pvc scale deployment --replicas=0 ubuntu 
deployment.apps/ubuntu scaled

test@(aks-cluster):~$ kubectl -n pvc describe pvc azure-disk-pvc | grep Events -A40
Events:
  Type     Reason              Age                 From           Message
  ----     ------              ----                ----           -------
  Warning  VolumeResizeFailed  60m (x13 over 60m)  volume_expand  error expanding volume "cvp/azure-disk-pvc" of plugin "kubernetes.io/azure-disk": azureDisk - disk resize is only supported on Unattached disk, current disk state: Attached, already attached to  /subscriptions/111111111-1111-111-11111-1111111/resourceGroups/MC_rg-test_test/providers/Microsoft.Compute/virtualMachineScaleSets/aks-default-31009192-vmss/virtualMachines/aks-default-31009192-vmss_0

test@(aks-cluster):~$ kubectl -n pvc describe pvc azure-disk-pvc 
Name:          azure-disk-pvc
Namespace:     cvp
StorageClass:  azure-disk
Status:        Bound
Volume:        pvc-f73f8774-ce41-43df-9ec3-ca7791cec457
Labels:        <none>
Annotations:   pv.kubernetes.io/bind-completed: yes
               pv.kubernetes.io/bound-by-controller: yes
               volume.beta.kubernetes.io/storage-provisioner: kubernetes.io/azure-disk
               volume.kubernetes.io/storage-resizer: kubernetes.io/azure-disk
Finalizers:    [kubernetes.io/pvc-protection]
Capacity:      2Gi
Access Modes:  RWO
VolumeMode:    Filesystem
Mounted By:    <none>
Conditions:
  Type                      Status  LastProbeTime                     LastTransitionTime                Reason  Message
  ----                      ------  -----------------                 ------------------                ------  -------
  FileSystemResizePending   True    Mon, 01 Jan 0001 00:00:00 +0000   Fri, 09 Apr 2021 15:51:01 +0300           Waiting for user to (re-)start a pod to finish file system resize of volume on node.
Events:                     <none>

test@(aks-cluster):~$ kubectl -n pvc scale deployment --replicas=1 ubuntu 

root@ubuntu-5fb979b685-xcvkl:/# df -h
Filesystem                                                                                                   Size  Used Avail Use% Mounted on
overlay                                                                                                       97G   43G   55G  44% /
tmpfs                                                                                                         64M     0   64M   0% /dev
tmpfs                                                                                                        7.9G     0  7.9G   0% /sys/fs/cgroup
/dev/sdd                                                                                                     7.9G   27M  7.8G   1% /azure-disk
/dev/sda1                                                                                                     97G   43G   55G  44% /etc/hosts
shm                                                                                                           64M     0   64M   0% /dev/shm
tmpfs                                                                                                        7.9G   12K  7.9G   1% /run/secrets/kubernetes.io/serviceaccount
tmpfs                                                                                                        7.9G     0  7.9G   0% /proc/acpi
tmpfs                                                                                                        7.9G     0  7.9G   0% /proc/scsi
tmpfs                                                                                                        7.9G     0  7.9G   0% /sys/firmware
root@ubuntu-5fb979b685-xcvkl:/# ^C
root@ubuntu-5fb979b685-xcvkl:/# exit
command terminated with exit code 130

test@(aks-cluster):~$ date
Fri 09 Apr 2021 15:52:47 PM EEST

test@(aks-cluster):~$ kubectl get sc azure-disk -o yaml | yq - d 'metadata.managedFields' | yq - d 'metadata.annotations'
allowVolumeExpansion: true
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  creationTimestamp: "2021-03-23T16:15:35Z"
  name: azure-disk
  resourceVersion: "3521"
  selfLink: /apis/storage.k8s.io/v1/storageclasses/azure-disk
  uid: e572daa3-74e4-4d3e-b21f-66029b939024
parameters:
  cachingmode: ReadOnly
  kind: Managed
  storageaccounttype: Standard_LRS
provisioner: kubernetes.io/azure-disk
reclaimPolicy: Delete
volumeBindingMode: Immediate

```

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
kubectl version 
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.3", GitCommit:"1e11e4a2108024935ecfcb2912226cedeafd99df", GitTreeState:"clean", BuildDate:"2020-10-14T12:50:19Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.7", GitCommit:"14f897abdc7b57f0850da68bd5959c9ee14ce2fe", GitTreeState:"clean", BuildDate:"2021-01-22T17:29:38Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}

- Size of cluster (how many worker nodes are in the cluster?)
10 worker
- General description of workloads in the cluster (e.g. HTTP microservices, Java app, Ruby on Rails, machine learning, etc.)
HTTP microservices

**Comment by andyzhangx** [comment]
pls scale replica to 0 and then resize pvc, that should take a few minutes.

**Comment by elops-od** [comment]
hi,
i have the same problem
maybe someone know workround ?

@andyzhangx 
>pls scale replica to 0 and then resize pvc, that should take a few minutes.

in the test results from the first message, you can see that before pvc resizing the number of replicas was set to 0
and it didn't help

**Comment by elkh510** [comment]
any updates ?

**Comment by miwithro** [comment]
@andyzhangx

**Comment by andyzhangx** [comment]
could be due to this issue: https://github.com/kubernetes/kubernetes/issues/94764#issuecomment-829150737, there is some corner cases that detach disk hang there and vm is in failed state until another vm update happened, disk detach succeeded.

**Comment by andyzhangx** [comment]
already provided the answer, this corner case issue was fixed on azure disk csi driver in v1.3.0
